src/tests/posix/memccpy_test.c

```diff
@@ -0,0 +1,174 @@
+/**
+ * @file
+ * @brief Tests for memccpy function
+ * 
+ * @author Giannis Zapantis  (github name: tech-gian)
+ * @date 18/3/2020
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <embox/test.h>
+
+
+EMBOX_TEST_SUITE("memccpy");
+
+
+/* TEST CASES for testing memccpy
+ * on strings
+ */
+
+
+TEST_CASE("empty static allocated string with character off the string") {
+    
+    char t1[] = "To be copied";
+    char t2[128];
+
+    void* test = memccpy(t2, t1, 'w', strlen(t1)+1);
+    test_assert_zero(strcmp(t2, t1));
+    test_assert_null(test);
+}
+
+
+TEST_CASE("non-empty static allocated string with character off the string") {
+
+    char t1[] = "To be copied";
+    char t3[] = "Before copy";
+
+    void* test = memccpy(t3, t1, 'w', strlen(t1)+1);
+    test_assert_zero(strcmp(t3, t1)); 
+    test_assert_null(test);
+}
+
+
+TEST_CASE("non-empty static allocated string with character on the string") {
+
+    char t1[] = "To be copied";
+    char t5[] = "Before copy";
+
+    void* test = memccpy(t5, t1, 'c', strlen(t1)+1);
+    test_assert_zero(strcmp(t5, "To be  copy"));
+    test_assert_not_null(test);
+}
+
+
+TEST_CASE("empty static allocated string with character on the string") {
+
+    char t1[] = "To be copied";
+    char t6[128];

```

Comment:
I agree with @anton-bondarev, the variables names are weird. In case if you want to use `t` then why `t1, t6` but not `t1, t2`?

Reply:
You are right. I thought that it would be better to have an order for the dest variables and keep the same t1 for the src variable. What names do you suggest to put instead?